### PR TITLE
podman: bump RLIMIT_NOFILE also without CAP_SYS_RESOURCE

### PR DESCRIPTION
**Pull request by giuseppe**
If we are not able to make arbitrary changes to the RLIMIT_NOFILE when
lacking CAP_SYS_RESOURCE, don't fail but bump the limit to the maximum
allowed.  In this way the same code path works with rootless mode.

Closes: https://github.com/containers/libpod/issues/2123

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

File: cmd/podman/main.go [hunk 1]
```diff
@@ -148,17 +148,17 @@ func main() {
...
+		if err := syscall.Setrlimit(syscall.RLIMIT_NOFILE, rlimits); err != nil {
+			if err := syscall.Getrlimit(syscall.RLIMIT_NOFILE, rlimits); err != nil {
+				return errors.Wrapf(err, "error getting rlimits")
+			}
+			rlimits.Cur = rlimits.Max
 			if err := syscall.Setrlimit(syscall.RLIMIT_NOFILE, rlimits); err != nil {
 				return errors.Wrapf(err, "error setting new rlimits")
 			}
-		} else {
-			logrus.Info("running as rootless")
```
Comment (TomSweeneyRedHat) [inline cmd/podman/main.go]: IDK what others think, but it might be nice to keep this in play.

**Comment by mheon** [comment]
I'd almost rather just ignore the error from setrlimit, but this is fine too.
/approve

File: cmd/podman/main.go (hunk 1 above)
Comment (giuseppe) [reply cmd/podman/main.go]: why do we need it if the more generic version handles both cases?

**Comment by TomSweeneyRedHat** [comment]
Changes LGTM, would like to keep the informational message, but a toss on that.   Tests don't look happy though.

File: cmd/podman/main.go (hunk 1 above)
Comment (TomSweeneyRedHat) [reply cmd/podman/main.go]: I think it's just a nicety/reminder when scrounging through the logs that the command was running rootless.  Not a strong leaning.

**Comment by mheon** [comment]
Code LGTM

**Pull request merged**